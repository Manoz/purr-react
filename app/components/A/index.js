import styled from 'styled-components';

const A = styled.a`
  color: black;
  text-decoration: none;
  text-decoration-skip: objects;
  border-bottom: 1px solid transparent;
  transition: color .15s, border .15s;

  &:link,
  &:visited {
    color: black;
  }

  &:hover {
    color: var(--primaryColor);
    border-bottom: 1px solid var(--primaryColor);
    transition: color .15s, border .15s;
  }

  &:focus {
    outline: 0;
  }
`;

export default A;
