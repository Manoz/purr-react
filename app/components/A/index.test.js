import React from 'react';
import { shallow } from 'enzyme';

import A from './index';

const renderComponent = (props = {}) => shallow(
  <A href="http://example.com" {...props}>
    test
  </A>,
);

describe('<A />', () => {
  it('should render an <a>', () => {
    const wrapper = renderComponent();
    expect(wrapper.type()).toEqual('a');
  });

  it('should have an href attribute', () => {
    const wrapper = renderComponent();
    expect(wrapper.prop('href')).toEqual('http://example.com');
  });

  it('should have children', () => {
    const wrapper = renderComponent();
    expect(wrapper.contains('test')).toEqual(true);
  });
});
