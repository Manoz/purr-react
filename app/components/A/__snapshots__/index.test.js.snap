// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should render an <a> 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "A-nontsf-0",
        "isStatic": true,
        "rules": Array [
          "padding-bottom:2px;color:#c9abf8;text-decoration:none;text-decoration-skip:objects;border-bottom:1px solid transparent;transition:color .15s,border .15s;&:link,&:visited{color:#c9abf8;}&:hover{color:var(--secondaryColor);border-bottom:1px solid var(--secondaryColor);transition:color .15s,border .15s;}&:focus{outline:0;}",
        ],
      },
      "displayName": "A",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "A-nontsf-0",
      "target": "a",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  href="http://example.com"
>
  test
</StyledComponent>
`;
